lib/kino/process.ex

```diff
@@ -581,21 +586,98 @@ defmodule Kino.Process do
     Map.put_new(rels, lookup, edge)
   end
 
-  defp traverse_links({rels, _idx, pid_keys}) do
+  defp traverse_links({rels, _idx, resource_keys}) do
     rels_with_links =
-      Enum.reduce(pid_keys, rels, fn {pid, _idx}, rels_with_links ->
-        {:links, links} = process_info(pid, :links)
+      Enum.reduce(resource_keys, rels, fn
+        {pid, _idx}, rels_with_links when is_pid(pid) ->
+          {:links, links} = process_info(pid, :links)
+
+          Enum.reduce(links, rels_with_links, fn link_pid, acc ->
+            add_new_links_to_acc(resource_keys, pid, link_pid, acc)
+          end)
 
-        Enum.reduce(links, rels_with_links, fn link_pid, acc ->
-          add_new_links_to_acc(pid_keys, pid, link_pid, acc)
-        end)
+        _, rels_with_links ->
+          rels_with_links
       end)
 
     rels_with_links
   end
 
-  defp add_new_links_to_acc(pid_keys, pid, link_pid, acc) do
-    case pid_keys do
+  defp maybe_traverse_ets_tables(supervision_tree_data, opts) do
+    if Keyword.get(opts, :render_ets_tables, false) do
+      do_traverse_ets_tables(supervision_tree_data)
+    else
+      supervision_tree_data
+    end
+  end
+
+  defp do_traverse_ets_tables(supervision_tree_data) do
+    active_ets_tables =
+      :ets.all()
+      |> Enum.map(fn table ->
+        table
+        |> :ets.info()
+        |> Map.new()
+      end)
+
+    ets_owner_map =
+      active_ets_tables
+      |> Enum.reduce(%{}, fn table_info, acc ->
+        case Map.get(table_info, :owner) do
+          owner_pid when is_pid(owner_pid) -> Map.put(acc, owner_pid, table_info)
+          _ -> acc
+        end
+      end)
+
+    ets_heir_map =
+      active_ets_tables
+      |> Enum.reduce(%{}, fn table_info, acc ->
+        case Map.get(table_info, :heir) do
+          heir_pid when is_pid(heir_pid) -> Map.put(acc, heir_pid, table_info)
+          _ -> acc
+        end
+      end)
```

Comment:
You can fold the ets table lookup and both reduces into a single traversal:

    Enum.reduce(:ets.all, {%{}, %{}}, fn ... ->

You could also keep a single map if you prefer and keep the keys namespaced, such as `{:heir, pid}` and `{:owner, pid}`. It will be less data to pass around.